client/fed_linux_test.go

```diff
@@ -918,8 +918,192 @@ func TestObjectPutNonRecursiveDirPath(t *testing.T) {
+	})
+
+	// Test deleting an existing object
+	t.Run("testDeleteObject", func(t *testing.T) {
+		objectToDeletePelicanUrl := fmt.Sprintf("pelican://%s/with-write/%s", discoveryUrl.Host, filepath.Base(objectToBeDeleted))
+		err = client.DoDelete(fed.Ctx, objectToDeletePelicanUrl, false, client.WithTokenLocation(tempToken.Name()))
+		require.NoError(t, err)
+
+		_, err = os.Stat(objectToBeDeleted)
+		if err != nil {
+			if os.IsNotExist(err) {
+				require.True(t, true, "Object does not exist as expected")
+			} else {
+				require.NoError(t, err, "Unexpected error occurred while checking object")
+			}
+		} else {
+			require.Fail(t, "Object should not exist but it does")
+		}
+	})
+
+	// Test deleting an empty collection
+	t.Run("testDeleteEmptyCollection", func(t *testing.T) {
+		emptyCollectionPelicanUrl := fmt.Sprintf("pelican://%s/with-write/%s", discoveryUrl.Host, filepath.Base(emptyCollectionToBeDeleted))
+		err := client.DoDelete(fed.Ctx, emptyCollectionPelicanUrl, false, client.WithTokenLocation(tempToken.Name()))
+		require.NoError(t, err)
+		_, err = os.Stat(emptyCollectionToBeDeleted)
+		if err != nil {
+			if os.IsNotExist(err) {
+				require.True(t, true, "Collection successfully deleted")
+			} else {
+				require.NoError(t, err, "Unexpected error occurred while checking collection")
+			}
+		} else {
+			require.Fail(t, "Collection should not exist but it does")
+		}
+	})
+
+	// Test deleting a non-empty collection without the recursive flag
+	t.Run("testDeleteNonEmptyCollectionWithoutRecursive", func(t *testing.T) {
+		collectionToDeletePelicanUrl := fmt.Sprintf("pelican://%s/with-write/%s", discoveryUrl.Host, filepath.Base(complexCollection))
+		err := client.DoDelete(fed.Ctx, collectionToDeletePelicanUrl, false, client.WithTokenLocation(tempToken.Name()))
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "cannot delete non-empty collection")
+
+		info, err := os.Stat(complexCollection)
+		require.NoError(t, err, "Error checking complex collection existence")
+		require.True(t, info.IsDir(), "Complex collection should exist but does not")
+	})
+
+	// Test deleting a non-empty collection with the recursive flag
+	t.Run("testDeleteNonEmptyCollectionWithRecursive", func(t *testing.T) {
+		collectionToDeletePelicanUrl := fmt.Sprintf("pelican://%s/with-write/%s", discoveryUrl.Host, filepath.Base(complexCollection))
+		err = client.DoDelete(fed.Ctx, collectionToDeletePelicanUrl, true, client.WithTokenLocation(tempToken.Name()))
+		require.NoError(t, err)
+		_, err = os.Stat(complexCollection)
+		if err != nil {
+			if os.IsNotExist(err) {
+				require.True(t, true, "Collection successfully deleted")
+			} else {
+				require.NoError(t, err, "Unexpected error occurred while checking collection")
+			}
+		} else {
+			require.Fail(t, "Collection should not exist but it does")
+		}
+	})
+
+	// Test attempting to delete an object in a prefix without writes capability and expecting a permission error
```

Comment:
Discussed offline: 

This check is done at the client itself: [link](https://github.com/PelicanPlatform/pelican/compare/main...Saartank:pelican:implement-object-delete#diff-39e4af22263d62c01d979fa1f6ec1770d58fc1b03b0aeec2f17ba84e7136c0dbR352-R355 )

So, there is no need to add additional test since there are no moving parts involved here.